permissions_validators/src/private_blockchain/query.rs

```diff
@@ -226,6 +226,8 @@ impl<W: WorldTrait> IsAllowed<W, QueryBox> for OnlyAccountsDomain {
                     ))
                 }
             }
+            #[cfg(not(feature = "roles"))]
+            _ => Err("Unable to compile with tests enabled, but without \"roles\", because it's a dev-dependency of `iroha_data_model`, and due to https://github.com/rust-lang/cargo/issues/6915".to_owned())
```

Comment:
Is this case reachable?

Reply:
No. But without this case, it won't compile. 

Basically, `#[cfg(test)]` pulls in all the `dev` dependencies. There are a few tests in `data_model` that require `roles` so when you compile `permissions_validators` in `test` mode, it compiles `iroha_data_model` with its `iroha_core` `"roles"` dependency, and so you need to have this exhaustive list of query values. 

Once we can specify feature dependencies in test mode, (the linked issue) we should remove this and make sure that when we compile without `roles`, then all crates are compiled without them.

Reply:
I see, thanks